server/src/main/java/org/candlepin/dto/api/v1/CertificateDTO.java

```diff
@@ -0,0 +1,167 @@
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.api.v1;
+
+import org.candlepin.dto.CandlepinDTO;
+
+import io.swagger.annotations.ApiModel;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+
+
+/**
+ * The CertificateDTO is a DTO representing all Candlepin certificates presented to the API.
+ */
+@ApiModel(parent = TimestampedCandlepinDTO.class, description = "DTO representing a certificate")
+public class CertificateDTO extends TimestampedCandlepinDTO<CertificateDTO> {
+    public static final long serialVersionUID = 1L;
+
+    protected String id;
+    protected String key;
+    protected String cert;
+    protected CertificateSerialDTO serial;
+
+
+    /**
+     * Initializes a new CertificateDTO instance with null values.
+     */
+    public CertificateDTO() {
+        // Intentionally left empty
+    }
+
+    public String getId() {
+        return this.id;
+    }
+
+    public CertificateDTO setId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    public String getKey() {
+        return this.key;
+    }
+
+    public CertificateDTO setKey(String key) {
+        this.key = key;
+        return this;
+    }
+
+    public String getCert() {
+        return this.cert;
+    }
+
+    public CertificateDTO setCert(String cert) {
+        this.cert = cert;
+        return this;
+    }
+
+    public CertificateSerialDTO getSerial() {
+        return this.serial;
+    }
+
+    public CertificateDTO setSerial(CertificateSerialDTO serial) {
+        this.serial = serial;
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        CertificateSerialDTO serial = this.getSerial();
+
+        return String.format("CertificateDTO [id: %s, key: %s, serial id: %s]",
+            this.getId(), this.getKey(), serial != null ? serial.getId() : null);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj instanceof CertificateDTO && super.equals(obj)) {
+            CertificateDTO that = (CertificateDTO) obj;
+
+            EqualsBuilder builder = new EqualsBuilder()
+                .append(this.getId(), that.getId())
+                .append(this.getKey(), that.getKey())
+                .append(this.getCert(), that.getCert())
+                .append(this.getSerial(), that.getSerial());
+
+            return builder.isEquals();
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        HashCodeBuilder builder = new HashCodeBuilder(37, 7)
```

Comment:
I believe so. I feel that if .equals is calling up to the parent, .hashCode should as well to keep up with the hashCodes being unequal if the inequality occurs in the parent.